tests/mock_virtual_machine.h

```diff
@@ -37,14 +37,28 @@ template <typename T = VirtualMachine,
 struct MockVirtualMachineT : public T
 {
     template <typename... Args>
-    MockVirtualMachineT(Args&&... args)
+    explicit MockVirtualMachineT(Args&&... args)
         : MockVirtualMachineT{std::make_unique<TempDir>(), std::forward<Args>(args)...}
     {
     }
 
     template <typename... Args>
-    MockVirtualMachineT(std::unique_ptr<TempDir>&& tmp_dir, Args&&... args)
+        requires(std::is_same_v<VirtualMachine, T>)
```

Comment:
This is totally up to you, but I thought I'd mention it anyway: you can also use `std::same_as` here, which is the concept form of `std::is_same_v`.